v-next/hardhat-build-system/src/internal/solidity/compiler/downloader.ts

```diff
@@ -17,15 +21,15 @@ import {
 } from "@nomicfoundation/hardhat-utils/fs";
 import debug from "debug";
 
-import { ERRORS } from "../../errors/errors-list.js";
-import { HardhatError, assertHardhatInvariant } from "../../errors/errors.js";
 import { download } from "../../utils/download.js";
 import { MultiProcessMutex } from "../../utils/multi-process-mutex.js";
 
 const log = debug("hardhat:core:solidity:downloader");
 
 const COMPILER_REPOSITORY_URL = "https://binaries.soliditylang.org";
 
+const ERRORS = HardhatError.ERRORS;
```

Comment:
This is purely personal, but I'd prefer to see HardhatError.ERRORS.SOME_ERROR instead of some variable whose origin is unclear. Feel free to ignore if you disagree.